# Peter Gordon's section of the code
## tasks:
- manipulate the data
- use two or three different models with hyperparameters

## outcome:
- to find what manipulation of the data will predict the final grade the best

In [6]:
#managing imports
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.neighbors import KNeighborsRegressor
from sklearn.tree import DecisionTreeRegressor
from sklearn.model_selection import train_test_split


In [15]:
#Loading the data
grade_base_data = pd.read_csv("Training_set.csv")

#setting the X, y, train and test
X = grade_base_data.iloc[:,:-1]
y = grade_base_data.iloc[:,-1]

#setting the train test split to 80/20
X_train, X_test, y_train, y_test = train_test_split(X,y, test_size=.2)

In [ ]:
#Peak at the data.
print(X)
print(y)

* At this point I have looked at the data
* The data is split up so that X is all data except the Final grade

In [20]:
#Setting up the Classifier clf as a KNN regressor with default parameters
clf = KNeighborsRegressor()
#Fitting the classifier with the training data
clf.fit(X_train,y_train)

#checking the score of the default KNN Regressor
print(f"Training score: {clf.score(X_train,y_train)}")
print(f"Testing score: {clf.score(X_test,y_test)}")

Training score: 0.5644413388680896
Testing score: 0.37337791105658735


In [57]:
#Setting up the Classifier clf as a KNN regressor Changing the parameters
clf = KNeighborsRegressor(n_neighbors= 5, weights="distance", algorithm="auto")
#Fitting the classifier with the training data
clf.fit(X_train,y_train)

#checking the score of the parameter KNN Regressor
print(f"Training score: {clf.score(X_train,y_train)}")
print(f"Testing score: {clf.score(X_test,y_test)}")

Training score: 0.8801448079370845
Testing score: 0.3512019891192997


In [67]:
# Setting up clf as a decision tree regressor default parameters
clf = DecisionTreeRegressor()
# fitting the data
clf.fit(X_train,y_train)

#checking the score of the default DT Regressor
print(f"Training score: {clf.score(X_train,y_train)}")
print(f"Testing score: {clf.score(X_test,y_test)}")

Training score: 0.8813494188025747
Testing score: 0.007548647795510455


## Outcome of changing the parameters was:
* Found that the best testing accuracy was never greater than 37% with knn regressor
* Training accuracy clearly overfit a little
* KNN may not be the best model

# Begining of data manipulation
